src/coreclr/src/vm/object.h

```diff
@@ -149,33 +149,17 @@ class Object
         m_pMethTab = pMT;
     }
 
-    VOID SetMethodTable(MethodTable *pMT
-                        DEBUG_ARG(BOOL bAllowArray = FALSE))
+    VOID SetMethodTable(MethodTable *pMT)
```

Comment:
>VOID SetMethodTable(MethodTable *pMT) [](start = 4, length = 37)

why get rid of the debug arg? its existence seems very intentional.

Reply:
This looks like a remnant from the times when array's method tables could be shared for arrays of reference type. It was important to not call this API by accident.
There is no sharing any more and we use the same API, regardless of the type of the object and whether it is an array or not. The check was still there, but not validating or protecting anything.

Reply:
I see; makes sense